### ISSUE TITLE
Doesn't show icon either context menu on Win10 when an "Open" program associated with .apk

### ISSUE THREAD

**Issue by AndroidDeveloperLB** [allangoing/apkshellext#78]
I've tried it, and it fails to show icons of APK files. 

Example attached here.

[app-release.zip](https://github.com/kkguo/apkshellext/files/3269291/app-release.zip)

Windows 10 .

**Comment by kkguo** [comment]
please re-download, I think it works now. feel free to reopen this if problem not fixed.

**Comment by AndroidDeveloperLB** [comment]
I've downloaded the latest today. From here:
".2.0.2 @June 10. 2019"

Still doesn't work. As I wrote here:
https://github.com/kkguo/apkshellext/issues/51#issuecomment-500880041

**Comment by AndroidDeveloperLB** [comment]
Maybe it's related to this:
https://github.com/kkguo/apkshellext/issues/62#issuecomment-500876673

?
Nowadays Google recommends using adaptive icon for all apps.
But maybe you have a fallback for this? 
Maybe it will use the normal icons when it has adaptive icon?

**Comment by kkguo** [comment]
I have your apk pasted above, and it's working on my side.
but yes, adaptive-icon is not supported yet.
please join the telegram discussion
https://t.me/joinchat/BrcZsQAtOKWeA7ThTyq3Ug

**Comment by AndroidDeveloperLB** [comment]
Please explain exactly what you did.

**Comment by kkguo** [comment]
I download your app-release.zip from first comment and rename as .apk. it shows icon for me.

**Comment by AndroidDeveloperLB** [comment]
What do you mean "rename as apk" ?
The APK file is inside the zip file.

**Comment by kkguo** [comment]
ah, ok, then I should unzipped it. but it works
![image](https://user-images.githubusercontent.com/485503/59450811-0c47df80-8dd8-11e9-8d2c-e60a9fe65484.png)

**Comment by blackcrack** [comment]
\*grin* sometimes must looking twice, like me too \*giggle*

**Comment by AndroidDeveloperLB** [comment]
But what about installing of the plugin?
What did you do to make it work?
Did you do more than what I tried?
Why do you see the icon and I don't ?

**Comment by blackcrack** [comment]
hmm.. maby did you not start the install.bat as Admin
i have make a exe for executing it as admin in the system..
you could open the admin-cmd and go into the folder, 
and execute the install.bat , i have extact all in one folder,
the langpacks in the own folders by side of the install.bat
and execute the install.bat as admin..
why as admin: if you have not a adminright's,
can the bat not copy the dll and langfile on the right place.

i have make a own installpack, with the extracted langfiles in the own folders and the binary for install as admin.., take a look (it is not offical) : 
http://blackysgate.de/files/folder-root/apkshellext2/

**Comment by kkguo** [comment]
> But what about installing of the plugin?
run install.bat, and it went through.
> What did you do to make it work?
I spend a lot of time to implement it. but you are not suppose to do that.
> Did you do more than what I tried?
same as above.
> Why do you see the icon and I don't ?
I don't know, maybe I'm just lucky enough.

May be you want to check this list:
https://github.com/kkguo/apkshellext/wiki/Quick-check-before-raise-issue

Please describe what you did, what is your OS? .net version? any error message? does your account has admin?
Normally people just run install.bat, authorize, and done.

BTW. 4 info-less questions in a row is not a good way to acquire help. You probably want to add what you think about your problem instead of asking what I did more than you tried. How I know what you have tried?? 
And I cannot cross check from different referenced topics and look for your comment to find what you did. I'm facing many topic. I expect to see you have clear statement here, or at least point me explicitly so that I can understand what you have seen and been through. "It doesn't work" doesn't work here at all.

**Comment by AndroidDeveloperLB** [comment]
Sorry for not giving more information. I though that this plugin is dead, and yet I saw new versions.
The issue is that the icons do not appear for APK files.
I've tested this on 2 PCs with Windows 10, on at the office and one at home.
Currently I'm on the PC at home, and this is its version:

![image](https://user-images.githubusercontent.com/5357526/59469729-54015200-8e3e-11e9-8b2f-3c1249b1f4e8.png)

And I have tried to run the installer batch file as admin. It didn't help.
There is no error message. The console shows all is fine.

Trying to install dot-net, I get this:

![image](https://user-images.githubusercontent.com/5357526/59469900-c40fd800-8e3e-11e9-88bd-1b2f774e92ef.png)


Here, attached a video showing what I try:

[2019-06-14_01-05-34.zip](https://github.com/kkguo/apkshellext/files/3288266/2019-06-14_01-05-34.zip)

**Comment by kkguo** [comment]
is it windows explorer or it's another file manager? Due to restriction of .NET, I cannot support other than windows file explorer.

**Comment by AndroidDeveloperLB** [comment]
It's windows explorer in dark theme, with 2 other plugins (that don't do anything to file icons):
- classic shell
- qttabbar

**Comment by AndroidDeveloperLB** [comment]
Is it possible that this plugin has issues because I've set a file association to APK files ? 
I've made opening it - to install the APK file, by using a batch file that is attached here .

[apkInstall.zip](https://github.com/kkguo/apkshellext/files/3289295/apkInstall.zip)

**Comment by blackcrack** [comment]
> 
> 
> It's windows explorer in dark theme, with 2 other plugins (that don't do anything to file icons):
> 
>     * classic shell
> 
>     * qttabbar
 
 ![ unbelievable, i give me a headshot ](http://www.blackysgate.de/files/folder-root/smilys/kolobok.us-1/am.gif) 
gnh.. they are not plugins, they are normal program's.. ! 
![it can not be ! ](http://www.blackysgate.de/files/folder-root/smilys/kolobok.us-0/headbang.gif) 

![oh no, no ,no ,no ... ](http://www.blackysgate.de/files/folder-root/smilys/kolobok.us-1/da.gif) ohw booy,
because this f.."§$"§$" M$ Windows learn the Peoples no more what's behind and be hold 
the users as "Dau Users" only, because they learn no more, what's real or not real.. ![old man say ](http://www.blackysgate.de/files/folder-root/smilys/kolobok.us-1/hg.gif)  it's need the time to comes a honest and sincare WinNT ordering system like Reactos !

> 
> 
> Is it possible that this plugin has issues because I've set a file association to APK files ?
> I've made opening it - to install the APK file, by using a batch file that is attached here .
> 
> [apkInstall.zip](https://github.com/kkguo/apkshellext/files/3289295/apkInstall.zip)

beeeppp !! wrong file.. this install not the Addon for the Filemanager Explorer
(the Browser for the Web names IExplorer for Internet-Explorer, so exploring the Folders and files or ,
explore the Internet for Internet-Exploring ..

@kkguo
look kkguo, this comes out, if you offer the whole packs as single packs, 
because you assume the users are smart enough, you should make a
msi or setup for the users with uninstall-possible

and maybe it's a setup more better where it is compressed 
with zip or something and is decompress able .. 


best regards
Blacky

**Comment by AndroidDeveloperLB** [comment]
No, they are plugins.
They change how the Explorer looks like, and they add more options to it. Tabs are the most useful feature for me, but they also have other stuff: search file name in current folder, preview of images by hovering...

See:

![image](https://user-images.githubusercontent.com/5357526/59502168-d113e200-8ea5-11e9-86a9-020d1eeca7cc.png)

The process is also still "explorer.exe"

**Comment by blackcrack** [comment]
it becomes offtopic, maybe is better to lock this @kkguo 

@AndroidDeveloperLB
these be addons (so add-on programs) for the explorer and starts/loaded extra by sides
of the explorer and become a extra loading.. and not by the Explorer, no, they have a own Prozess
and becomes a add in to the head program loaded.. and be not alone from "Explorer" loaded as read and plug in.. nope, they are started as a own instance .

**Comment by AndroidDeveloperLB** [comment]
Offtopic of what?
This plugin doesn't work. Still same topic.

"nope, they are started as a own instance ."
Incorrect. It's still the same process of the OS.
They are not stand alone. Both work at the same time, on the windows explorer itself.
They are not a shell of it.

**Comment by blackcrack** [comment]
if you can't understand a text-writed batch.. should you go into the school again and should not put the burden on @kkguo shoulders alone, this is not a issue alone at the Program, this is a issue at you, you should learn more about batch writing and understand like a system works.. support named not to learn all others like a computer & system works, nope, a issue is a task to repair bugs, but not by the peoples .. but at the programmed programm..
oky, this is the reason, why should be lock this conversation, because it is not a bug
on this programm, it's a bug some where else ..  ![understand ?](http://www.blackysgate.de/files/folder-root/smilys/kolobok.us-1/thumbup.gif) (question mark looking (if you have understand it ?))

**Comment by AndroidDeveloperLB** [comment]
Why all the toxic sentences? Why no manners?
Why not provide a helpful information instead of "go into school" ? 
Would you like me to disable those plugins to prove it's not because of them?

**Comment by blackcrack** [comment]
it's not my task to schooling you in Software.. :smile_cat:

**Comment by AndroidDeveloperLB** [comment]
BTW, just to prove you are wrong, there are other plugins that are similar to apkshellext that work just fine.
One of them is for showing content of SVG, and another is of WebP.
I have installed them both, and indeed they show their content when I download such files.
See:


![image](https://user-images.githubusercontent.com/5357526/59505387-f278cc00-8ead-11e9-884f-893b6e76dfb1.png)

**Comment by blackcrack** [comment]
they named ?

**Comment by AndroidDeveloperLB** [comment]
You don't have to "school" anyone and you don't have to behave like a kid in school . It's your choice of how to behave. It's your choice of how much time to waste instead of being neutral. You've wasted so much of your time thinking how to bash what I wrote instead of thinking about the problem.
You can just act as an adult instead and have a decent conversation. If you want more information you can ask. If you think something is wrong you can write it down. But saying I should go to school doesn't help anyone.

Those plugin names are:
svg explorer extension 0.1.1 . I think I got this from : https://github.com/maphew/svg-explorer-extension/releases
WebP Codec for Windows 0.19 (by Google) . This I think I got from: https://storage.googleapis.com/downloads.webmproject.org/releases/webp/index.html

**Comment by blackcrack** [comment]
cool thanks :+1: for the meaningful

https://www.webmproject.org/tools/
https://storage.googleapis.com/downloads.webmproject.org/releases/webm/index.html

**Comment by AndroidDeveloperLB** [comment]
I hope you now understand that all of those are not stand alone apps.
They are all addons/plugins/extensions or whatever you want to call them, but they are not stand-alone apps.

**Comment by blackcrack** [comment]
boy, i be more as 20 years in the net..
and have support in the past also the factory's 
with freebe software to build this world up like is now in Europe ..

i guess, this tread can close and lock now ..

**Comment by AndroidDeveloperLB** [comment]
Again, not related to the issue that was described, and not contributing to it either.

**Comment by kkguo** [comment]
Hey, guys, don't be personal please!
@AndroidDeveloperLB , apkshellext is writen in .NET, due to the platform restriction, it may not work like other shell extension. To be honest, .net is not good for shell extension at all, I chose it just because I don't want to go through WFC again.
So can you please dump the registry key for .apk class? like in the self-debug guide? I do see either icon doesn't work or menu doesn't work, first time I see a case both doesn't work at all. I'm quite interesting in this. 
If possible, can you please uninstall other plug-in and check if it becomes work? or if possible, can you use a fresh installed windows (or a VM) to check if it works for you?
And thanks for project you paste, I will think about migrate the code to C native to give better support for various windows setup.

**Comment by kkguo** [comment]
@blackcrack , I was trying to keep this as lite & green as possible, and I don't have good experience with package installer. I appreciate your help to make it an installer, if you can, please do regularly update, you could create a branch or another project for your delivery, I can also promote your work if you can provide support.
BTW, I don't know how to lock.

**Comment by kkguo** [comment]
> BTW, just to prove you are wrong, there are other plugins that are similar to apkshellext that work just fine.
> One of them is for showing content of SVG, and another is of WebP.
> I have installed them both, and indeed they show their content when I download such files.
> See:
> 
> ![image](https://user-images.githubusercontent.com/5357526/59505387-f278cc00-8ead-11e9-884f-893b6e76dfb1.png)

@AndroidDeveloperLB , after thinking about your scenario, I think I understand the issue now.
firstly, refer to this: https://stackoverflow.com/questions/49302484/whats-the-truth-on-shell-extensions-in-net
If you are not familiar with .NET, I give you a brief, also my understanding might not right.
So .NET is running in a managed environment on top of windows. But this environment is sealed, by saying sealed I mean .NET has to be called by a .NET process, also .NET process can call non-.net (non-managed) process but not the other way. Explorer is using a compatible layer to call .NET process.
I think your plug-in is running another window on top of native file explorer, and it's a WFC process, which is native and non-managed, they cannot call a COM wrapped by .NET even all the registry looks fine.
Actually apkshellext doesn't work with other file manager even they can show other shell extension (like Total Commander) . I'm surprised by Chrome as I think it's writen in native C but it works well with apkshellext.

Forgot to summary, I think apkshellext will not work with your plug-ins. especially the tab explorer. 
I will add a comment in the README.

**Comment by blackcrack** [comment]
@kkguo http://www.blackysgate.de/files/folder-root/apkshellext2/for%20kkguo/

and, how mus stay the files if it's installed, i have not so looking on the batchfile, mus they are stay by side, so can i installing the files in the system and registering the dll ?
i mus looking on the files more and looking on the batch more.. or it's possible 
to move the whole thing include lang-subdirectorys in a install-folder under programms\apkshell2\*dll + /langfiles/*.* ..
at moment have i less time, but i look what can i do.. :package:  :)

best regards
Blacky

**Comment by AndroidDeveloperLB** [comment]
@kkguo Please explain what to do exactly. It happens with all APK files that I try. Even a new APK of an Android project.
It's not just of the sample APK file that I've attached. 
As for a virtual machine, I could put a Linux one, as I did in the past (I usually tried out Linux for fun, using VirtualBox app for Windows), but for Windows I never did it. 
Would creating a new user account on Windows suffice, instead of uninstalling stuff or a virtual machine?

As for .Net, I am familiar with developing for it (I had a course for it a very long time ago), but I didn't go further with it, because I got a job in Android development and it didn't seem like .Net will be a good way to develop apps for Windows on smartphones.

Can't you make the plugin work as it should, by the rules? Maybe make a part of it in .Net, and a part in C/C++ ? 

So anyway, I don't know the technical details, but just ask for something, and I will help. If it involves privacy concerns, please let's use email instead.
Please explain about "dump the registry key for .apk class? like in the self-debug guide?" .

BTW, maybe you could work with this guy about parsing the APK to get the VectorDrawable:
https://github.com/hsiafan/apk-parser

As for adaptive icons, they are now widely used, and in the future it will not be required to have alternative icons resources, so apps could have only those.
Adaptive icons are just 2 layers of icons, one on top of the other. If you can handle one icon, I think you should be able to handle 2... 
I might be wrong though, as it could be more complicated than I think.
But I think the VectorDrawable could be much harder.

"And thanks for project you paste, I will think about migrate the code to C native to give better support for various windows setup.
"
Which project?

**Comment by kkguo** [comment]
@blackcrack so, the position of the dll doesn't matter, all you have to do to install is run 
regasm.exe /codebase apkshellext2.dll
but before this, installer suppose to get the admin authority and find regasm.exe position, that's what install.bat is doing, you probably want to uninstall first to release the file handler.
The language files has to locate in the same folder within a language name folder, it not decided by me, it's Microsoft.

**Comment by kkguo** [comment]
@AndroidDeveloperLB ,
This has how to dump the registry, but you don't have to do it now, I think they should look right.
https://github.com/kkguo/apkshellext/wiki/Quick-check-before-raise-issue#1-general-install-issue

A new account should work, as long as it doesn't load your other plug-ins.

Thanks for the reference to apk-parser, looks like it does extract information from manifest.xml, but doesn't really draw the drawable. And it's java, I cannot reuse its code directly. And I believe my parser is faster, I'm only extracting necessary information.

My own parser has ability to get the vectordrawable xml, I'm currently writing a render class to draw the vectors, as the xml itself is not a standard SVG, I cannot call an existing SVG library. 

Thanks for the project of svg explorer extension, it give me example of how to do it in Qt, and I like Qt.

**Comment by blackcrack** [comment]
@kkguo 
ah okey, so a normal install in 
ProgramFiles=C:\Program Files (x86)
ProgramFiles(x86)=C:\Program Files (x86)
ProgramW6432=C:\Program Files
it's no matter , i must just find a installer who checks the Architecture..


eehh whitch architecture is the apkshellext, x86 or the x86_64 (with 64bit enhanceed ?)
if ApkShellext2 32 bit only, can i use the standart-folder %ProgramFiles%/Apkshellext2
did not really looking on via Totalcmd lister-plugin.. could/should i am do :\

a modern installer must support the selecting of Administratorgroup, 
otherwish works a installer (also for copy files in system) not really :)

i must find only the right installer...

qt is not bad, if it's not be a huge codeball and later a big, heavy wight program as binarys,
but you must ever look to hold the whole slim ..

**Comment by AndroidDeveloperLB** [comment]
Why close the issue though?
I've tried creating a new account on Windows, and install apkshellext there.
It didn't help. Still have the same issue, of not seeing the app icon.
Please explain why this occurs, and what I can do.
Do you want me to uninstall the plugins and see if it helps?

**Comment by AndroidDeveloperLB** [comment]
@kkguo 
Can't you somehow use the code of the Android framework and the support library?
For example, I think `VectorDrawableCompat.createFromXmlInner` could be useful. I never tried it though.

**Comment by kkguo** [comment]
> @kkguo
> Can't you somehow use the code of the Android framework and the support library?
> For example, I think `VectorDrawableCompat.createFromXmlInner` could be useful. I never tried it though.

oh, it's a Java class, I cannot use the code directly. apkshellext is written in c# and running under .NET.

yes, please try uninstall your plugins if possible, I believe it's compatibility issue, and I cannot resolve it in short time.

**Comment by AndroidDeveloperLB** [comment]
Well one of the reasons I re-checked this plugin (it didn't work over the past year for me) is that I was hoping to find a solution for getting APK information purely via InputStream, because of this:

https://stackoverflow.com/q/56309165/878126

Anyway, I've removed both plugins of Explorer, restarted the OS, and installed apkshellext with admin, restarted explorer using the batch file too, and... still nothing.
I have APK files on the desktop itself. None of them got an icon shown. So I created a new APK on some folder. Still nothing.

![image](https://user-images.githubusercontent.com/5357526/59880355-7e5e8c80-93b5-11e9-8666-cb6aedeb5c7e.png)

I even restarted the whole OS. Still nothing.
Attached here what I get from "General install issue" section (from [here](https://github.com/kkguo/apkshellext/wiki/Quick-check-before-raise-issue#1-general-install-issue)) :

[General install issue.zip](https://github.com/kkguo/apkshellext/files/3312207/General.install.issue.zip)


Context menu doesn't show anything special:

![image](https://user-images.githubusercontent.com/5357526/59883416-b073ec80-93bd-11e9-8a9b-9e839cdafa23.png)

Attached here the file of "Context menu doesn't work" :

[Context menu doesn't work.zip](https://github.com/kkguo/apkshellext/files/3312215/Context.menu.doesn.t.work.zip)


Are you absouloutly sure that it's not because I have a file association with APK already, of a batch file to install it?

**Comment by kkguo** [comment]
Regarding your stackoverflow question, I'm not familiar with Android platform, and I'm not sure if you can get a unzipped stream for original file stream can be unzipped, but if you can, you could find enough information from manifest.xml, it's binary compiled, but the format is quite easy. This is actually what apkshellext is doing.

I didn't see issue with your dumped files. looks like it's a real issue, and I'm reopening this and rephrased the title.
Can you please do one small experiment? with clean environment (without other plugins), install with install.bat, check any apk file on desktop or explorer, go back to apkshellext.dll, try remove it or rename. If you can do that, means windows doesn't load up the dll, and the issue is in register process. if you can not, means windows has loaded the dll, and the issue happens during COM query or apk parsing stage.

**Comment by AndroidDeveloperLB** [comment]
You mean install and trying to delete the `apkshellext.dll` file?
OK.
I've tried now. I get this error:

![image](https://user-images.githubusercontent.com/5357526/60383444-50461000-9a79-11e9-8d37-8605acfd5d69.png)

However, after running `uninstall.bat`, I still can't delete it. I also tried to call `restart_explorer.bat` , and it still didn't let me delete it.
After a real restart of the OS, I succeed deleting the file.

**Comment by kkguo** [comment]
So some other application may load and lock the dll as well (eg: Chrome)
You may find some powerful tool (such as Process explorer) to find which app is locking the dll.

WIndows 10 seems not only querying registry when loading shell extension, it's doing more and some times will ignore registry settings. Because we are running .NET, it's even more complicated. I need to read more.

**Comment by AndroidDeveloperLB** [comment]
@kkguo Why would some other app load and lock the dll of this addon, and what is the chance of this to happen only after I launch "install.bat" ? 

Anyway, I now tried V2.0.6 on the laptop at the office.
I launched "install.bat" with admin and then run "restart_explorer.bat" , and it still doesn't show the APK icon.

And when I tried to remove the dll file (and in fact each of the dll files), even without running "uninstall.bat" file, it let me remove it.

I need to check it at home too. I wonder if I will see same results.
Shouldn't the dll file/s be locked when it says it succeeded to install?

Please explain what should I try to do to help fix those issues. I liked this addon when it used to work for me. 

Watch attached video:

[2019-07-17_12-56-55.zip](https://github.com/kkguo/apkshellext/files/3401199/2019-07-17_12-56-55.zip)

**Comment by kkguo** [comment]
When Chrome downloaded a file, it will call windows api to get the icon. so the extension dll is load and locked.
And also for same reason, after restart_explorer, the dll is still locked by other app. You need to end all the processes which loaded the dll in order to release the file handler.
You can delete the dll only because nothing has loaded the dll.

If I understand correctly, your windows doesn't load the dll, but Chrome loaded the dll.
This is interesting.

**Comment by AndroidDeveloperLB** [comment]
But Chrome didn't download a dll file. It downloaded a zip file, and I decompressed it via 7-zip.
The dll is supposed to be locked if it's used. 
What I've found is not that it's locked before using the "install", after just decompressing or downloading anything. it's locked after I ran it.
If you wanted, I could just close Chrome or any other web browser that I use, to show that it's not related.

And now the behavior is different. It's never locked. None of the dll files are locked, ever.

"If I understand correctly, your windows doesn't load the dll, but Chrome loaded the dll."
How do you understand that anything loaded the dll? It's not locked. I can now delete the files whenever I wish.

Anyway, tested on the PC at home. 
Got same results. I can delete the dll files after "install.bat" was launched, even with admin.

**Comment by kkguo** [comment]
So install.bat will not let windows lock the DLL, it only register the DLL, DLL locks when Windows or Chrome or any other app use the DLL functions via COM interface, i.e. show APK icon.
Chrome load the DLL if you download an APK file. 
If you still using the tab thing in your file explorer, I recommend you try to put the apk file on desktop, it should load the DLL too.

**Comment by AndroidDeveloperLB** [comment]
I don't understand the new plan. 
Please explain in steps.
Also, why do you keep talking about Chrome? I can kill it and still have the same issue.

**Comment by kkguo** [comment]
Just want to make sure we are at the same page.
Apkshellext doesn't work on your computer, it doesn't show up icon nor context menu for apk files. At earlier time, somehow the dll was locked and you cannot delete them until restart computer.  But now you don't see the lock issue anymore.
Here is my assumption and inference:
The dll locked because windows or something else loaded the dll, because it's a .net dll, it will be remained in the memory for a while. This also means the registration and query of the dll is working. Associate the fact that you have the "tab window" plugin which covers the native file explorer (I believe that's how it works), I guess apkshellext actually is working with file explorer but not your "tab window" plugin just because they are not compatible. You could try put apk files directly on the desktop, it may show up icons.
Regarding Chrome, I normally see the dll lockup by Chrome when I debug. It is also possible the dll was locked by other applications, I have no idea about that. But one more hint, even you exit Chrome from the right-button-menu on the icon on task bar, it may still runs in backgroud, use Task Manager to kill them all.

**Comment by blackcrack** [comment]
(@kkguo, by the way, here works well in win7pro (i have updated it to 2.0.7) and no problems so far.. the once problem what i see is the unregister dll, where must restart the explorer and then should it possible for remove it.. *imho* or just reboot and then should it be possible to remove the whole folder... best regards Blacky )

**Comment by kkguo** [comment]
@blackcrack Thanks for confirm, yes, a managed DLL need to be released by its caller to be deleted, and in order to let explorer release, you have to restart it. I did't see any other way to workaround.
I think a possible solution is, if you want to make a installer, create different folder for different version, and ask user to restart computer to delete the old version.

**Comment by AndroidDeveloperLB** [comment]
@kkguo OK here's what I did:
1. Uninstalled both the plugins I use.
2. Restarted the PC
3. Installed apkshellext (2.7) with admin, without running any web browser.
4. Put some APK files on the desktop
5. Restarted the PC.

Still. Nothing.

Can you please do what I talked about before?
To set a file association to APK ? I have a file association to APK files to a batch file that has this content in it:

```
@echo %1
c:
cd/
cd "C:\Users\user\AppData\Local\Android\Sdk\platform-tools"
adb.exe install -r %1


```
Attached here the batch file.

[apkInstall.zip](https://github.com/kkguo/apkshellext/files/3414000/apkInstall.zip)

**Comment by blackcrack** [comment]
@kkguo warning, let check the apk's with more virusscaners !

**Comment by AndroidDeveloperLB** [comment]
@blackcrack Was this about me?
I already proved that it has nothing to do with any web browser and any plugin that I have.
I was just suggesting that maybe this issue is because of what I did - associate APK file to something that can operate on it.
Maybe when there is a file association, the plugin fails to show an icon.

**Comment by kkguo** [comment]
@AndroidDeveloperLB , sorry I misunderstood your term about "file association", by that I think you mean "open with". 
Yes, after chose your bat as the default app to open .apk file, all the icon lost. But actually context menu IS STILL WORKING. See the image I pasted. So that I still don't know if your problem is the same.
![image](https://user-images.githubusercontent.com/485503/61646113-6f9a1c80-ac76-11e9-8a53-f2f835b5e574.png)

To fix this, you have to delete this key from registry:
Computer\HKEY_CURRENT_USER\Software\Microsoft\Windows\CurrentVersion\Explorer\FileExts\.apk\UserChoice
of course, this will remove your "file association" bat file from apk. I think there is a way to let them co-exist, but please try first to see if it resolves your problem.

BTW, I'm going to add adb function into the context menu. although it has been on the list for quite a while...

**Comment by AndroidDeveloperLB** [comment]
Removing the registry key made the icon appear as it used to be.
However, I don't want to only have an icon. I use APK files a lot, and thus when I double click on them, I chose it will automatically install the APK file.

And it shouldn't bother each other. Just like I have addons for showing SVG and WEBP in thumbnails/icons, and yet I can open them fine on editor apps.

As for the context menu that you've shown, I can't reproduce it.
After making the file-association that I have, the context menu of apkshellext is gone.

See attached video:

[2019-07-23_00-24-55.zip](https://github.com/kkguo/apkshellext/files/3419207/2019-07-23_00-24-55.zip)

Can you please add a file association , and have a setting of what to do upon opening the file (one is to install it, maybe another is to install and try to launch the app ) ?

**Comment by kkguo** [comment]
I have a workaround for you, go to
HKEY_LOCAL_MACHINE\SOFTWARE\Classes\\.apk, 
check the (Default) key value, on windows10, normally it is apk.1, if it's not, please replace apk.1 in next registry path with the name you see.
Go to or create following key, 
HKEY_LOCAL_MACHINE\SOFTWARE\Classes\apk.1\shell\open
(Default) = "Open"
HKEY_LOCAL_MACHINE\SOFTWARE\Classes\apk.1\shell\open\command
(Default) = "Your command path"
Then you should see the "Open" command again.

**Comment by AndroidDeveloperLB** [comment]
@kkguo I don't understand. Can you please make a video about it?

**Comment by kkguo** [comment]
@AndroidDeveloperLB I don't have a way for video, you need to open your registry with regedit.
If you don't understand this, please google it.
I'm giving a solution, and I'm closing this now.

**Comment by AndroidDeveloperLB** [comment]
@kkguo I didn't mean that I don't know how to use the registry. I mean that your instructions are unclear.
And why close this issue if it still exists?
I only have a file association. It doesn't mean the icon shouldn't appear...
It works fine with other addons that I have...
WEBP, SVG...
As for an app to do screen recording (and screenshots) , I can suggest you about what I use:
https://getsharex.com/

**Comment by kkguo** [comment]
@AndroidDeveloperLB, I understand your concern, there is issue, and it's better to be fixed. 
But please also understand this is not a commercial software, and nothing has to be perfect, I have a list of priority other than this issue and you are welcome if you can fix it by yourself. And if you resolve it, please also let the community know.
Issue got closed because it has a solution. it maybe not the best, but it should not bother for this moment.
And I don't know how other project doing, please understand my limitation regarding ability and time.
You get what you get. period.

**Comment by kkguo** [comment]
I have a new wiki page for your issue, maybe helpful, I encourage you try it out yourself and input what you think might helpful to the wiki page.
https://github.com/kkguo/apkshellext/wiki/Known-issues-and-solution#3-if-you-have-a-customized-open-operation-it-will-disables-icon-and-menu

**Comment by AndroidDeveloperLB** [comment]
@kkguo This doesn't exist, so I can't delete it:

>  2) remove key HKEY_CURRENT_USER\Software\Microsoft\Windows\CurrentVersion\Explorer\FileExts.apk

![image](https://user-images.githubusercontent.com/5357526/62077586-1bcd9b80-b253-11e9-8883-56cf26d040a4.png)

I also tried the "3. If you have a customized "Open" operation, it will disables icon and menu." part, but again this is unclear. Here's what I got after trying it :

[2019-07-30_00-37-10.zip](https://github.com/kkguo/apkshellext/files/3444335/2019-07-30_00-37-10.zip)

And both don't work. It doesn't show the icons of the APK files.
